espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Router.java

```diff
@@ -124,10 +125,7 @@
 
 
     @SuppressWarnings("unchecked")
-    public BaseResponse route(IHTTPSession session) {
-        String uri = session.getUri();
-        Method method = session.getMethod();
-
+    public synchronized BaseResponse route(String uri, Method method, Map<String, String> parms,  Map<String, String> files) {
```

Comment:
having synchronized methods is usually expensive for performance. ReentrantLock is much better alternative to it. One just have to check carefully where to guard shared resources